stl/inc/typeindex

```diff
@@ -38,9 +38,17 @@ public:
         return *_Tptr == *_Right._Tptr;
     }
 
+#if !_HAS_CXX20
     _NODISCARD bool operator!=(const type_index& _Right) const noexcept {
         return !(*this == _Right);
     }
+#else // ^^^ !_HAS_CXX20 / _HAS_CXX20 vvv
+    _NODISCARD strong_ordering operator<=>(const type_index& _Right) const noexcept {
+        return *_Tptr == *_Right._Tptr      ? strong_ordering::equal
+             : _Tptr->before(*_Right._Tptr) ? strong_ordering::less
+                                            : strong_ordering::greater;
+    }
+#endif // !_HAS_CXX20

```

Comment:
Style: We conventionally prefer to avoid negated conditionals (when doing so doesn't complicate the code). Here, the if/else can be swapped, so we can positively test `_HAS_CXX20`.